src/index.test.tsx

```diff
@@ -0,0 +1,98 @@
+import {fireEvent, render, renderHook, screen} from '@testing-library/react';
+import {ClickOutsideProvider, useClickOutside} from './index';
+import { UseClickOutsideHandlerProps } from "./types";
+
+const onOutsideClick = jest.fn();
+
+afterEach(() => jest.clearAllMocks());
+
+const renderUseClickOutside = (ref: UseClickOutsideHandlerProps['ref'], disabled?: boolean) => renderHook(() =>
+  useClickOutside({
+    ref,
+    onOutsideClick,
+    disabled
+  })
+);
+
+const renderClickOutsideProvider = (options?: boolean | AddEventListenerOptions) => render(
+  <div>
+    <button onMouseDown={e => e.stopPropagation()}>outside</button>
+    <ClickOutsideProvider onOutsideClick={onOutsideClick} options={options}>
+      <button>child</button>
+    </ClickOutsideProvider>
+  </div>
+);
+
+describe('useClickOutside', () => {
+  it('should call onOutsideClick when clicked outside ref element', () => {
+    const ref = {current: document.createElement('div')};
+
+    renderUseClickOutside(ref)
+
+    fireEvent.mouseDown(document);
+
+    expect(onOutsideClick).toHaveBeenCalled();
+  });
+
+  it('should not call onOutsideClick when clicked inside ref element', () => {
+    const ref = {current: document.createElement('div')};
+    const innerElement = document.createElement('span');
+    ref.current.appendChild(innerElement);
+
+    renderUseClickOutside(ref)
+
+    fireEvent.mouseDown(innerElement);
+
+    expect(onOutsideClick).not.toHaveBeenCalled();
+  });
+
+  it('should not call onOutsideClick when disabled is true', () => {
+    const ref = {current: document.createElement('div')};
+
+    renderUseClickOutside(ref,true)
```

Comment:
```suggestion
    renderUseClickOutside(ref, {disabled: true})
```
More explicit & easier to consume without need to look into definition of this function